[quant][graphmode] insert_quant_dequant work with qconfig_dict

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25370 [refactor] Remove unnecessary checks in InsertQuantDeQuantImpl
* #25281 [quant][graphmode] Skip inserting observers for Tensors inside fused op
* #25217 [quant][graphmode] Integration tests for qconfig_dict
* **#25127 [quant][graphmode] insert_quant_dequant work with qconfig_dict**
* #25069 [graphmode] insert_observers use qconfig_dict

Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
[quant][graphmode] insert_quant_dequant work with qconfig_dict

Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

ZolotukhinM reviewed (changes requested):
I think we should not duplicate the logic from `insertObservers` for figuring out what values we need to quantize. Instead we should just look for the corresponding `CallMethod` nodes and replace them with Q-DQ nodes. This logic was introduced earlier, but earlier it was slightly better "justified" than now.

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

ZolotukhinM reviewed (approved):
Yeah, it's fine if you do it in a separate PR.

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)

jerryzh168 committed:
Update on "[quant][graphmode] insert_quant_dequant work with qconfig_dict"


Summary:
Extend insert_quant_dequant pass to go through forward function graphs

Test Plan:
```
python test/test_jit.py 'TestJit.test_insert_quant_dequant'
python test/test_quantizer.py
```

Reviewers:
pt1quant
Subscribers:

Tasks:

Tags:

Differential Revision: [D17001137](https://our.internmc.facebook.com/intern/diff/D17001137)
